### PR TITLE
feat: altered plan table

### PR DESCRIPTION
**Pull request by Adewale66**
## How this PR should be tested

The endpoint is protected and only a user with admin role can access it.

# new DB schema

```mermaid
erDiagram
    Plan {
        varchar(255) id PK
        varchar(255) name
        int price
        int duration
        varchar(255) description
        varchar(255) duration_unit
        varchar(255) features
    }
```

## How to test

- access route without being authenticated.
- create a plan with a user without admin role.
- create a plan that already exists.

### Checklist of what was implemented

- [x] Created endpoint (POST /api/v1/plans) to create new subscription plans.
- [x]  Validated creation of plans.
- [x]  Validated only users with admin roles can access
- [x] Wrote unit tests for creating plans.
- [x] Wrote unit tests for duplicate plan creation.

## Reference issue
[[FEAT] Implement API Endpoints for Creating Subscription Plans](https://github.com/hngprojects/hng_boilerplate_java_web/issues/6)

### CODE REVIEW

**Review by Am0du — DISMISSED** [review]
LGTM

**Review by lurldgbodex — APPROVED** [review]

File: src/main/resources/db/migration/V7.1__alter_plan_table.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE plans
```
Comment (lurldgbodex) [inline src/main/resources/db/migration/V7.1__alter_plan_table.sql:1]: please check the application and update the version numbver

**Review by lurldgbodex — CHANGES_REQUESTED** [review]
check the app and update your version number to be a new later version number

**Comment by Adewale66** [comment]
The version number is correct, it avoids the mistake of using another number someone else has already used and it's just a table update @Am0du even agreed with it.

**Comment by lurldgbodex** [comment]
There are newer version which has run on the database. when your application runs on the database, it will throw an error even though there is no 7.1 because of reference issue.You can try it on your local

**Review by lurldgbodex — APPROVED** [review]

**Pull request merged**